server/src/main/java/org/elasticsearch/TransportVersions.java

```diff
@@ -256,11 +258,9 @@ public static NavigableMap<Integer, TransportVersion> getAllVersionIds(Class<?>
             }
         }
 
-        return Collections.unmodifiableNavigableMap(builder);
-    }
+        Collections.sort(definedTransportVersions);
 
-    static Collection<TransportVersion> getAllVersions() {
-        return VERSION_IDS.values();
+        return Collections.unmodifiableList(definedTransportVersions);
```

Comment:
nit: unmodifiableList creates a wrapper which makes every access indirect. Instead, use `List.copyOf` to get an unmodifiable list from another collection.